### PR TITLE
feat(host): flag to use req.host instead of localhost

### PR DESCRIPTION
**Pull request by infoxicator**
In order for the browser tests to run the dev-cdn the module map has to point to the host instead of localhost. I have added the flag `useHost` to be set to true when running browser tests and false as default when using one-app-runner so it points to localhost during local development.

Tested by passing the flag from one-app, building a dev docker image, passing the image local to one-app-runner and running the browser tests. both scenarios worked as expected

### CODE REVIEW

**Comment by nellyk** [comment]
Do we need to mention something in the Readme?

File: src/index.js [hunk 1]
```diff
@@ -57,6 +57,7 @@ export default ({
   remoteModuleMapUrl,
   useLocalModules,
   appPort,
+  useHost,
 }) => {
```
Comment (JAdshead) [inline src/index.js:61]: what about adding something like below here. you can probably come up with a better name however
```suggestion
}) => {
  const hostAddress = useHost ? `http://${req.headers.host}` : `http://localhost:${process.env.HTTP_ONE_APP_DEV_CDN_PORT}`;
```

File: src/index.js [hunk 2]
```diff
@@ -106,7 +107,7 @@ export default ({
             const remoteModuleMap = JSON.parse(r.body);
 
             const { modules } = remoteModuleMap;
-            const oneAppDevStaticsAddress = `http://localhost:${process.env.HTTP_ONE_APP_DEV_CDN_PORT}/static`;
+            const oneAppDevStaticsAddress = useHost ? `http://${req.headers.host}/static` : `http://localhost:${process.env.HTTP_ONE_APP_DEV_CDN_PORT}/static`;
```
Comment (JAdshead) [inline src/index.js:110]: continued from comment above -
```suggestion
            const oneAppDevStaticsAddress = `${hostAddress}/static`;
```
this change can also be applied below

File: src/index.js [hunk 3]
```diff
@@ -141,7 +142,7 @@ export default ({
         : {},
       (useLocalModules ? getLocalModuleMap({
         pathToModulemap: path.join(localDevPublicPath, 'module-map.json'),
-        oneAppDevCdnAddress: `http://localhost:${process.env.HTTP_ONE_APP_DEV_CDN_PORT}`,
+        oneAppDevCdnAddress: useHost ? `http://${req.headers.host}}` : `http://localhost:${process.env.HTTP_ONE_APP_DEV_CDN_PORT}`,
```
Comment (JAdshead) [inline src/index.js:145]:
```suggestion
        oneAppDevCdnAddress: hostAddress,
```

File: src/index.js (hunk 1 above)
Comment (infoxicator) [reply src/index.js:58]: had to put it inside the `get` for the `req` to be accessible but Implemented your changes here: https://github.com/americanexpress/one-app-dev-cdn/pull/14/commits/1c5a17c9779e2264f2584d27afc1031b0fdfd53b

File: src/index.js (hunk 2 above)
Comment (infoxicator) [reply src/index.js]: https://github.com/americanexpress/one-app-dev-cdn/pull/14/commits/1c5a17c9779e2264f2584d27afc1031b0fdfd53b

File: src/index.js (hunk 3 above)
Comment (infoxicator) [reply src/index.js]: https://github.com/americanexpress/one-app-dev-cdn/pull/14/commits/1c5a17c9779e2264f2584d27afc1031b0fdfd53b

**Comment by infoxicator** [comment]
> Do we need to mention something in the Readme?

Added documentation to README https://github.com/americanexpress/one-app-dev-cdn/pull/14/commits/1c5a17c9779e2264f2584d27afc1031b0fdfd53b

File: README.md [hunk 4]
```diff
@@ -84,13 +84,23 @@ location where the remote module map is located (i.e. `https://my-domain.com/map
 
 Type: `boolean`
 
-Default: `true`<br>
-Whether to use modules from `localDevPublicPath`. Passed as `true` or `false`, defaults to `true`.
+Default: `false`<br>
```
Comment (infoxicator) [inline README.md:87]: In the code this never defaulted to true so I updated the README to reflect

**Pull request merged**